[V3] mypy not running properly in CI

I'm seeing the following output in CI in the v3 branch, under the `run mypy` stage:
```
Run hatch run test:run-mypy
─────────────────────────── test.py3.10-1.24-minimal ───────────────────────────
/bin/sh: 1: mypy: not found
Error: Process completed with exit code 127.
```

Nonetheless, GHA reports this stage as succeeding :upside_down_face: Maybe because 127 is not one of the exit codes it considers a failure. However I think we can consider this test a failure. How should we fix this?

cc @dstansby

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -43,7 +43,3 @@ jobs:
     - name: Run Tests
       run: |
         hatch env run --env test.py${{ matrix.python-version }}-${{ matrix.numpy-version }}-${{ matrix.dependency-set }} run
-    - name: Run mypy
-      continue-on-error: true
-      run: |
-        hatch run test:run-mypy

